### PR TITLE
bpo-39019: Implement missing __class_getitem__ for os.DirEntry

### PR DESCRIPTION
**Pull request by isidentical**
https://bugs.python.org/issue39019

### CODE REVIEW

File: Misc/NEWS.d/next/Library/2019-12-10-21-22-20.bpo-39019.SkVoG0.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+Implement ``__class_getitem__`` for ``os.DirEntry``.
```
Comment (ilevkivskyi) [inline Misc/NEWS.d/next/Library/2019-12-10-21-22-20.bpo-39019.SkVoG0.rst]:
```suggestion
Implement dummy ``__class_getitem__`` for ``os.DirEntry``.
```

File: Modules/posixmodule.c [hunk 2]
```diff
@@ -12808,6 +12815,7 @@ static PyMethodDef DirEntry_methods[] = {
     OS_DIRENTRY_STAT_METHODDEF
     OS_DIRENTRY_INODE_METHODDEF
     OS_DIRENTRY___FSPATH___METHODDEF
+    {"__class_getitem__", DirEntry_cls_getitem, METH_O|METH_CLASS, NULL},
```
Comment (ilevkivskyi) [inline Modules/posixmodule.c:12818]: Would it be possible to use Argument Clinic for this (to be consistent with the rest of the code in this module)?

**Review by ilevkivskyi — COMMENTED** [review]
Thanks! Couple comments also here.

File: Modules/posixmodule.c (hunk 2 above)
Comment (isidentical) [reply Modules/posixmodule.c:12818]: I dont know if it possible, contextlib also uses argument clinic but they seperated `__class_getitem`. I thought there is a catch about this but I can try to do this with argument clinic.

**Comment by isidentical** [comment]
Closing since pep 585 accepted

**Comment by gvanrossum** [comment]
Hm, `os.DirEntry[str]` still gives an error. Maybe it still needs a `__class_getitem__` method?

**Comment by isidentical** [comment]
I was going to send a new PR but couldnt see it on PEP 585

**Pull request closed without merging**